keycloak_add_users.py

```
@@ -0,0 +1,209 @@
+        json_response = response.json()
+        self.AccessToken = json_response['access_token']
+
+    def Get(self, url, body, headers):
+        headers["Authorization"] = "Bearer {}".format(self.AccessToken)
+        return self.rawMethod("GET", url, body, headers)
+
+    def Post(self, url, body, headers):
+        headers["Authorization"] = "Bearer {}".format(self.AccessToken)
+        return self.rawMethod("POST", url, body, headers)
+
+    def Put(self, url, body, headers):
+        headers["Authorization"] = "Bearer {}".format(self.AccessToken)
+        return self.rawMethod("PUT", url, body, headers)
+
+    def putUser(self, realm_name, user):
+        headers = {"Content-type": "application/json"}
+        body = user.toJson()
+        response = self.Put("/auth/admin/realms/{}/users/{}".format(realm_name, user.id), body, headers)
+
+    def doesUserExist(self, realm_name, requestedUser):
+        return_dict = dict()
+        headers = {}
+        response = self.Get("/auth/admin/realms/{}/users/".format(realm_name), "", headers)
+        for user in response.json():
+            if user['username'] == requestedUser.username:
+                return_dict['user'] = user
+                return_dict['bool'] = True
+                return return_dict
+
+        return_dict['user'] = requestedUser
+        return_dict['id'] = 0
+        return_dict['bool'] = False
+        return return_dict
+
+    def rawMethod(self, method, url, body, headers):
+        fullUrl = self.BaseURL+url
+        try:
+            if (method == "GET"):
+                r = requests.get(fullUrl, headers=headers)
+            elif (method == "POST"):
+                r = requests.post(fullUrl, data=body, headers=headers)
+            elif (method == "PUT"):
+                r = requests.put(fullUrl, data=body, headers=headers)
+            else:
+                print("Not supported at this time")
+                exit(1)
```

Exit code of 127 is more usual here

---

On second thoughts if you are capturing exceptions, in order to exit, you should probably just raise a new exception.